src/mpl/src/mem/mpl_trmem_avx.c

```diff
@@ -16,8 +16,10 @@
 #include <string.h>
 #include <immintrin.h>
 
+#ifndef __clang__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdiscarded-qualifiers"
+#endif
```

Comment:
Can we remove the `$pragma` altogether? I.e. fix the warning by maybe adding a cast?